src/node.rs

```diff
@@ -929,10 +935,12 @@ impl<K, V> TreeBin<K, V> {
     /// method.
     pub(crate) unsafe fn defer_drop_without_values<'g>(
         bin: Shared<'g, BinEntry<K, V>>,
-        guard: &'g Guard,
+        guard: &'g Guard<'_>,
     ) {
-        guard.defer_unchecked(move || {
-            if let BinEntry::Tree(mut tree_bin) = *bin.into_owned().into_box() {
+        guard.retire(bin.as_ptr(), |mut link| {
+            let bin = link.cast::<BinEntry<K, V>>();
```

Comment:
I'm very surprised that `.cast` requires no `unwrap` or `unsafe` :thinking:

Reply:
It is `unsafe` :thinking:

Reply:
But.. There's no `unsafe` block here? So how then can you call it?

Reply:
Resolved in discussion below.